src/main/scala/analysis/MemoryModelMap.scala

```diff
@@ -254,20 +522,125 @@ class MemoryModelMap {
       logRegion(range, region)
     }
   }
+
+  def mergeRegions(regions: Set[MemoryRegion]): MemoryRegion = {
+    // assert regions are of the same type
+    regions.foreach(uf.makeSet)
+    regions.foreach(uf.union(regions.head, _))
+    uf.find(regions.head)
+  }
+
+  private def returnRegion(region: MemoryRegion): MemoryRegion = {
+    uf.find(region)
+  }
+
+  private def returnRegion(region: StackRegion): StackRegion = {
+    uf.find(region.asInstanceOf[MemoryRegion]).asInstanceOf[StackRegion]
+  }
+
+  private def returnRegion(region: DataRegion): DataRegion = {
+    uf.find(region.asInstanceOf[MemoryRegion]).asInstanceOf[DataRegion]
+  }
+
+  private def returnRegion(region: HeapRegion): HeapRegion = {
+    uf.find(region.asInstanceOf[MemoryRegion]).asInstanceOf[HeapRegion]
+  }
+
+  def getHeap(directCall: DirectCall): HeapRegion = {
+    require(directCall.target.name == "malloc", "Should be a malloc call")
+    heapCalls(directCall)
+  }
+
+  def getStack(allocationSite: CFGPosition): Set[StackRegion] = {
+    stackAllocationSites.getOrElse(allocationSite, Set.empty).map(returnRegion)
+  }
+
+  def getData(cfgPosition: CFGPosition): Set[DataRegion] = {
+    cfgPositionToDataRegion.getOrElse(cfgPosition, Set.empty).map(returnRegion)
+  }
+
+  def addMergeRegions(regions: Set[MemoryRegion], name: String): Unit = {
+    mergedRegions(regions) = name
+  }
+
+  def getMergedName(regions: Set[MemoryRegion]): String = {
+    mergedRegions(regions)
+  }
 }
 
 trait MemoryRegion {
   val regionIdentifier: String
+  val start: BigInt
+  val subAccesses: mutable.Set[BigInt] = mutable.Set()
 }
 
-case class StackRegion(override val regionIdentifier: String, start: BitVecLiteral, parent: Procedure) extends MemoryRegion {
-  override def toString: String = s"Stack($regionIdentifier, $start, ${parent.name})"
+case class StackRegion(override val regionIdentifier: String, start: BigInt, parent: Procedure) extends MemoryRegion {
+  override def toString: String = s"Stack($regionIdentifier, $start, ${parent.name}, $subAccesses)"
 }
```

Comment:
Why hasn't a size been added to this?

Reply:
`start` also doesn't override the trait's `start` variable because you haven't made it `override val`. This applies to the other region case classes too.